api/src/main/java/org/openmrs/module/kenyaemr/calculation/library/hiv/LostToFollowUpCalculation.java

```diff
@@ -86,26 +89,26 @@ public CalculationResultMap evaluate(Collection<Integer> cohort, Map<String, Obj
 			EncounterType hivEnrolmentEncounter = encounterService.getEncounterTypeByUuid(HivMetadata._EncounterType.HIV_ENROLLMENT);
 			Encounter lastHivEnrollmentEncounter = EmrUtils.lastEncounter(patientService.getPatient(ptId), hivEnrolmentEncounter);
 			// Is patient alive and in HIV program
+			
+			// Check patient latest appointment and get the appointment date
+			String sql = "select max(start_date_time) as appointment_date from patient_appointment" + " where patient_id =" + ptId + " and appointment_service_id = 1;";
+			String lastAppointmentDate = Context.getAdministrationService().executeSQL(sql, true).get(0).get(0).toString();
```

Comment:
What does this row return for a patient with no appointments of service id = 1? please try it out so that we avoid any possible NPE

Reply:
Fixed